crates/net/network/src/transactions.rs

```diff
@@ -1169,6 +1170,52 @@ impl TransactionFetcher {
     }
 }
 
+/// Sample a random subset of consecutive elements in range `0..total_count`.
+#[derive(Debug, Default)]
+struct RangeSampler {
+    /// The total number elements in the range.
+    total_count: usize,
+    /// The starting index of the sample range.
+    sample_start: usize,
+    /// The end index of the sample range.
+    sample_end: usize,
+}
+
+impl RangeSampler {
+    pub fn new(total_count: usize, sample_count: usize) -> Self {
+        assert!(sample_count <= total_count);
+        if total_count == 0 {
+            return Self { total_count: 0, sample_start: 0, sample_end: 0 };
+        }
+
+        let mut rng = rand::thread_rng();
+        let sample_start = rng.gen_range(0..total_count);
+        let sample_end = sample_start.checked_add(sample_count).expect("should not overflow");
```

Comment:
same here, no panics in release mode, please use `wrapping_add` instead

Reply:
this means there needs to be
```rust
struct RangeSampler {
    /// The starting index of the sample A range, incase `sample_a_start` + `sample_count` wrapped. Otherwise, 
    // just use sample B.
    sample_a_start: Option<usize>,
    /// The starting index of the sample B range.
    sample_b_start: usize,
    /// The end index of the sample B range.
    sample_b_end: usize,
}
```

Reply:
*sample_a_end (since if it wraps, we know it started with 0).

feel free to do it any other way arithmetically, as long as `wrapping_add` is used